public/app/core/utils/explore.test.ts

```diff
@@ -440,3 +458,28 @@ describe('when buildQueryTransaction', () => {
     expect(transaction.request.interval).toEqual('2h');
   });
 });
+
+describe('generateEmptyQuery', () => {
+  it('should generate query with dataSourceOverride and without queries', async () => {
+    const query = await generateEmptyQuery([], 1, { type: 'loki', uid: 'ds1' });
+
+    expect(query.datasource?.uid).toBe('ds1');
```

Comment:
I think `?` is not needed here. Also in other places.

```suggestion
    expect(query.datasource.uid).toBe('ds1');
```

Reply:
According to TypeScript it is needed. If I removed it I'd get an error:
<img width="405" alt="Screenshot 2023-01-24 at 17 25 39" src="https://user-images.githubusercontent.com/48948963/214349918-18c182d9-6551-43df-bd56-30c974ee832b.png">